client/state/plans/selectors.js

```diff
@@ -40,35 +35,16 @@ export const getPlan = createSelector(
 );
 
 /**
- * Returns a plan price object
+ * Returns a plan price
  * @param  {Object}  state     global state
  * @param  {Number}  productId the plan productId
  * @param  {Boolean} isMonthly if true, returns monthly price
- * @return {Object}  a plan price object with the following shape:
- * {String} currencySymbol: '$',
- * {String} decimalMark: '.',
- * {String} dollars: '8',
- * {String} cents: '25'
+ * @return {Number}  plan price
  */
-export function getPlanPriceObject( state, productId, isMonthly = false ) {
+export function getPlanRawPrice( state, productId, isMonthly = false ) {
 	const plan = getPlan( state, productId );
-	if ( ! plan || ! plan.formatted_price || ! plan.raw_price ) {
+	if ( ! plan || ! plan.raw_price ) {
```

Comment:
As `plan.raw_price` can be `0` for the free plan, this condition would be wrongly `true`. I suggest checking for `! plan.hasOwnProperty( 'raw_price' )`.